[bitnami/jasperreports] Add support for an existing secret

mr-onion-2 opened the pull request:
### Description of the change

Add support for an existingSecret value

### Checklist

<!-- [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

corico44 commented:
Hello @mr-onion-2,
Could you tell me how you have tested that the new parameter _existingSecret_ works correctly to be able to reproduce it?
Also, when adding a new attribute and not fixing a bug but adding a small new functionality, it would be more appropriate to consider the change as a new minor and not as a patch. So the chart version should be changed to **14.4.0.**

mr-onion-2 commented:
Hi @corico44

Thanks for the feedback. I've updated the chart version and also fixed an oversight where the existing secret was assumed to be have the default name.

Please see some tests below. I hope these are sufficient.

**1. Deploy - without specifying an existing secret**
```
helm install test bitnami/jasperreports -n bitnami-test --create-namespace --set mariadb.auth.rootPassword=password --set mariadb.auth.password=password
-> Success

kubectl describe pod test-jasperreports-89784f669-nn7nn -n bitnami-test | grep JASPERREPORTS_PASSWORD
JASPERREPORTS_PASSWORD: <set to the key 'jasperreports-password' in secret 'test-jasperreports'>  Optional: false
-> Using chart secret, as expected
```

**2. Upgrade - without specifying the jasperreportsPassword value**
```
helm upgrade test bitnami/jasperreports -n bitnami-test --create-namespace --set mariadb.auth.rootPassword=password --set mariadb.auth.password=password
-> Fails, as expected (_"You must provide your current passwords when upgrading the release."_)
```

**3. Upgrade - specifying the jasperreportsPassword value**
```
export JASPERREPORTS_PASSWORD=$(kubectl get secret --namespace "bitnami-test" test-jasperreports -o jsonpath="{.data.jasperreports-password}" | base64 -d)

helm upgrade test bitnami/jasperreports -n bitnami-test --create-namespace --set mariadb.auth.rootPassword=password --set mariadb.auth.password=password --set jasperreportsPassword=$JASPERREPORTS_PASSWORD
-> Success
```

**4. Deploy - with an existing secret**
```
helm uninstall test -n bitnami-test

kubectl create secret generic my-jasperreport-secret --from-literal=jasperreports-password=password -n bitnami-test

helm install test bitnami/jasperreports -n bitnami-test --create-namespace --set mariadb.auth.rootPassword=password --set mariadb.auth.password=password --set existingSecret=my-jasperreport-secret
-> Success

kubectl describe pod test-jasperreports-89784f669-7fxtd -n bitnami-test | grep JASPERREPORTS_PASSWORD
JASPERREPORTS_PASSWORD: <set to the key 'jasperreports-password' in secret 'my-jasperreport-secret'>  Optional: false
-> Using the new secret, as expected
```

**5. Upgrade - without specifying the jasperreportsPassword value or existing secret**
```
helm upgrade test bitnami/jasperreports -n bitnami-test --create-namespace --set mariadb.auth.rootPassword=password --set mariadb.auth.password=password
-> Fails, as expected (_"You must provide your current passwords when upgrading the release."_)
```

**6. Upgrade - specifying the existing secret value**
```
helm upgrade test bitnami/jasperreports -n bitnami-test --create-namespace --set mariadb.auth.rootPassword=password --set mariadb.auth.password=password --set existingSecret=my-jasperreport-secret
-> Success
```

mr-onion-2 commented:
Got my branch in a bit of a pickle trying to rebase and sign off the commits 

Will re-open new PR
